### ISSUE TITLE
[browser calls]: requests timing do not match with client ones

### ISSUE THREAD

**Issue by legomushroom** [Glimpse/Glimpse.Client.Hud#157]
Requests timing in do not much with client ones:

![hud-timings-issue](https://user-images.githubusercontent.com/1478800/27711481-13ce4e8c-5ce9-11e7-91dc-8510918df0ff.gif)

**Comment by mike-kaufman** [comment]
Need to verify if this is a 304 case, where the HUD is incorrectly showing the timing data for the original 200.

**Comment by davidkpiano** [comment]
This isn't a 304 case, at least based on my initial understanding.

What's happening is that there are _two_ separate proxies - the `Glimpse.Browser.Agent` (which is the one reported) and a separate, ad-hoc proxy that _tries_ to do the same thing in parallel in the actual HUD.

Because there are two separate proxies, both time measurements are completely independent, so it makes sense that there would be a discrepancy between timings.

I don't believe there is a short-term fix to this. The long-term solution would be to provide hooks in `Glimpse.Browser.Agent` that the HUD can directly pull from, in order to get the _same_ reported timing without having to do the timing measurement itself.

**Comment by philliphoff** [comment]
Closing in favor of #175.
